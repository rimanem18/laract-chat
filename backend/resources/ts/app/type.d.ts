export type PromiseState = 'idle' | 'loading' | 'rejected'

export type Message = {
  id: number
  group_id: number
  user_id: number
  name: string
  content: string
  created_at: string
  roles: string[]
}
export type Group = {
  id: number
  name: string
  roles: string[]
}
export type Role = {
  id: number
  name: string
  color: string
}
export type RoleUser = {
  user_id: number
  role_id: number
}
export type RoleGroup = {
  group_id: number
  role_id: number
}

export type Groups = {
  byId: Record<string, Group>
  allIds: string[]
}

export type Roles = {
  byId: Record<string, Role>
  allIds: string[]
}

export type Messages = {
  byId: Record<string, Message>
  allIds: string[]
}

/**
 * Payloads
 *
 * バックエンドから送られてくるデータの型
 */
export type GroupsPayload = {
  public_groups: {
    id: number
    name: string
  }[]
  private_groups: {
    id: number
    name: string
  }[]
  role_group: {
    role_id: number
    group_id: number
  }[]
}
export type RoleGroupPayload = {
  role_group: {
    group_id: number
    role_id: number
  }[]
}
export type RolesPayload = {
  id: number
  name: string
  color: string
}
export type MessagePayload = {
  id: number
  group_id: number
  content: string
  created_at: string
  user_id: number
  name: string
}
export type RoleUserPayload = {
  user_id: number
  role_id: number
}

// Slice
export type AuthState = {
  name: string
  email: string
  password: string
  promise: PromiseState
  message: string
}
export type GroupsState = {
  groups: Groups
  roleGroup: {
    byId: Record<string, RoleGroup>
    allIds: string[]
  }
  oldestId: number
  promise: PromiseState
}
export type MenuState = {
  isOpen: boolean
}
export type PostState = {
  userId: number
  content: string
  promise: PromiseState
}
export type RolesState = {
  roles: Roles
  promise: PromiseState
}
export type UserState = {
  id: number
  name: string
  email: string
  roles: string[]
  promise: PromiseState
}
export type ChatMessagesState = {
  messages: {
    byId: Record<string, Message>
    allIds: string[]
  }
  roles: {
    byId: Record<string, Role>
    allIds: string[]
  }
  roleUser: {
    byId: Record<string, RoleUser>
    allIds: string[]
  }
  promise: PromiseState
}
