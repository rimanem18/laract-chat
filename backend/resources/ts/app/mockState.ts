import { RootState } from './store'

export const mockState: RootState = {
  authSlice: {
    name: '太郎',
    email: 'taro@example.com',
    password: 'authpassword',
    promise: 'idle',
    message: 'auth message',
  },
  chatMessagesSlice: {
    ids: ['message1', 'message2'],
    entities: {
      message1: {
        id: 1,
        name: 'hoge',
        content: 'fugafuga',
        created_at: '1900-01-01',
      },
      message2: {
        id: 2,
        name: 'piyo',
        content: 'fugafuga',
        created_at: '1900-01-01',
      },
    },
    promise: 'idle',
  },
  postSlice: {
    userId: 1,
    content: 'fugafuga',
    promise: 'idle',
  },
  userSlice: {
    id: 1,
    name: '太郎',
    email: 'taro@example.com',
    promise: 'idle',
  },
}
