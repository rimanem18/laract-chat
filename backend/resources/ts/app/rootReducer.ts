import { combineReducers } from '@reduxjs/toolkit'
import chatMessagesSlice from '../features/ChatMessagesSlice'
import userSlice, { authApi } from '../features/UserSlice'

const rootReducer = combineReducers({
  chatMessagesSlice,
  userSlice,
})

export type RootState = ReturnType<typeof rootReducer>
export default rootReducer
