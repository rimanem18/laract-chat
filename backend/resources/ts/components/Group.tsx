import { List, ListItemButton, ListItemText } from '@mui/material'
import React, { useEffect, useState } from 'react'
import { useHistory } from 'react-router-dom'
import {
  useAppDispatch,
  useGroupsEntities,
  useGroupsIds,
  useMenuIsOpen,
  useParamGroupId,
} from '../app/hooks'
import { toggleMenuOpen } from '../slices/MenuSlice'
import AddGroupModal from './AddGroupModal'

const Group = () => {
  const dispatch = useAppDispatch()
  const groupIds = useGroupsIds()
  const groupsEntities = useGroupsEntities()

  return (
    <>
      <AddGroupModal />
      <List
        sx={{
          '&::-webkit-scrollbar': {
            width: 2,
            borderRadius: 5,
          },
          '&::-webkit-scrollbar-track': {
            boxShadow: `inset 0 0 6px rgba(0, 0, 0, 0.3)`,
          },
          '&::-webkit-scrollbar-thumb': {
            backgroundColor: 'rgba(0, 0, 0, 0.3)',
            outline: `1px solid rgba(0, 0, 0, 0.3)`,
          },

          height: '65vh',
          overflowY: 'scroll',
          overflow: 'none',
        }}
      >
        {groupIds.map((id: string) => {
          return (
            <GroupItem
              key={id}
              id={groupsEntities[id].id.toString()}
              name={groupsEntities[id].name}
            />
          )
        })}
      </List>
    </>
  )
}
export default React.memo(Group)

/**
 * Components
 */
type GroupItemProps = {
  id: string
  name: string
}
const GroupItem = React.memo(({ id, name }: GroupItemProps) => {
  const activeGroupId = useParamGroupId()
  const history = useHistory()
  const [isActive, setIsActive] = useState(false)
  const dispatch = useAppDispatch()

  useEffect(() => {
    setIsActive(activeGroupId === id ? true : false)
  }, [activeGroupId])

  const goTo = () => {
    history.push(`/groups/${id}`)
    dispatch(toggleMenuOpen(false))
  }

  return (
    <>
      <ListItemButton
        selected={isActive}
        onClick={goTo}
        key={id}
        data-testid={`group${id}`}
      >
        <ListItemText primary={name}></ListItemText>
      </ListItemButton>
    </>
  )
})
