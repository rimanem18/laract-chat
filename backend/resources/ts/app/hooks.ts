import React, { useCallback, useEffect, useMemo, useState } from 'react'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import type { RootState, AppDispatch } from './store'

import {
  chatMessagesPromiseSelector,
  messageContentSelector,
  messageDatetimeSelector,
  messageGroupIdSelector,
  messageNameSelector,
  messageRoleColorSelector,
  selectMessages,
} from '../selectors/ChatMessagesSelector'
import {
  userEmailSelector,
  userIdSelector,
  userNameSelector,
  userPromiseSelector,
  userRoleNumberIdsSelector,
  userRolesSelector,
} from '../selectors/UserSelector'
import {
  authEmailSelector,
  authMessageSelector,
  authNameSelector,
  authPasswordSelector,
  authPromiseSelector,
} from '../selectors/AuthSelector'
import {
  postContentSelector,
  postPromiseSelector,
  postUserIdSelector,
} from '../selectors/PostSelector'
import {
  groupNameSelector,
  groupPathSelector,
  groupsDefaultPathSelector,
  groupsOldestIdSelector,
  groupsPromiseSelector,
  selectGroups,
  selectRoleGroup,
} from '../selectors/GroupsSelector'
import { rolesPromiseSelector, selectRoles } from '../selectors/RolesSelector'
import { useParams } from 'react-router'
import { menuIsOpenSelector } from '../selectors/MenuSelector'

// プレーンな useDispatch と useSelector の代わりにアプリ全体で使用する
export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

// AuthSlice
export const useAuthState = () => {
  const name = useAppSelector(authNameSelector)
  const email = useAppSelector(authEmailSelector)
  const password = useAppSelector(authPasswordSelector)
  const message = useAppSelector(authMessageSelector)
  const promise = useAppSelector(authPromiseSelector)

  const authState = {
    name,
    email,
    password,
    message,
    promise,
  }
  return authState
}

// User Slice
export const useUserState = () => {
  const id = useAppSelector(userIdSelector)
  const email = useAppSelector(userEmailSelector)
  const name = useAppSelector(userNameSelector)
  const roles = useAppSelector(userRolesSelector)
  const roleNumberIds = useAppSelector(userRoleNumberIdsSelector)
  const promise = useAppSelector(userPromiseSelector)

  const userState = {
    id,
    email,
    name,
    roles,
    roleNumberIds,
    promise,
  }
  return userState
}

// Groups Slice
export const useGroupsState = () => {
  const groups = {
    byId: useAppSelector(selectGroups.byId),
    allIds: useAppSelector(selectGroups.allIds),
    allNumberIds: useAppSelector(selectGroups.allNumberIds),
  }
  const roleGroup = {
    byId: useAppSelector(selectRoleGroup.byId),
    allIds: useAppSelector(selectRoleGroup.allIds),
  }
  const promise = useAppSelector(groupsPromiseSelector)
  const oldestId = useAppSelector(groupsOldestIdSelector)
  const defaultPath = useAppSelector(groupsDefaultPathSelector)

  const groupState = {
    groups,
    roleGroup,
    promise,
    oldestId,
    defaultPath,
  }
  return groupState
}

export const useGroupName = (id: string) => {
  const nameFactory = useAppSelector(groupNameSelector)
  const name = useMemo(() => {
    return nameFactory(id)
  }, [nameFactory, id])

  return name
}
export const useGroupPath = (id: string) => {
  const pathFactory = useAppSelector(groupPathSelector)
  const path = useMemo(() => {
    return pathFactory(id)
  }, [pathFactory, id])

  return path
}

// Menu Selector
export const useMenuIsOpen = () => {
  return useAppSelector(menuIsOpenSelector)
}

// ChatMessages Slice
export const useChatMessagesState = () => {
  const messages = {
    byId: useAppSelector(selectMessages.byId),
    allIds: useAppSelector(selectMessages.allIds),
  }
  const promise = useAppSelector(chatMessagesPromiseSelector)

  const chatMessagesState = {
    messages,
    promise,
  }
  return chatMessagesState
}

export const useMessageName = (id: string) => {
  const nameFactory = useAppSelector(messageNameSelector)
  const name = useMemo(() => {
    return nameFactory(id)
  }, [nameFactory, id])

  return name
}
export const useMessageContent = (id: string) => {
  const contentFactory = useAppSelector(messageContentSelector)
  const content = useMemo(() => {
    return contentFactory(id)
  }, [contentFactory, id])

  return content
}
export const useMessageDatatime = (id: string) => {
  const datetimeFactory = useAppSelector(messageDatetimeSelector)
  const datetime = useMemo(() => {
    return datetimeFactory(id)
  }, [datetimeFactory, id])

  return datetime
}
export const useMessageRoleColor = (id: string) => {
  const roleColorFactory = useAppSelector(messageRoleColorSelector)
  const roleColor = useMemo(() => {
    return roleColorFactory(id)
  }, [roleColorFactory, id])

  return roleColor
}

export const useMessageGroupId = (id: string) => {
  const groupIdFactory = useAppSelector(messageGroupIdSelector)
  const groupId = useMemo(() => {
    return groupIdFactory(id)
  }, [groupIdFactory, id])

  return groupId
}

export const usePostState = () => {
  const userId = useAppSelector(postUserIdSelector)
  const content = useAppSelector(postContentSelector)
  const promise = useAppSelector(postPromiseSelector)

  const postState = {
    userId,
    content,
    promise,
  }
  return postState
}

export const useRolesState = () => {
  const roles = {
    byId: useAppSelector(selectRoles.byId),
    allIds: useAppSelector(selectRoles.allIds),
  }
  const promise = useAppSelector(rolesPromiseSelector)

  const rolesState = {
    roles,
    promise,
  }

  return rolesState
}

// Render
export const useScrollToBottom = (
  refObject: React.MutableRefObject<HTMLElement | null>
) => {
  const el = refObject.current
  if (el !== null) {
    el.scrollTo(0, el.scrollHeight)
  }
}

/**
 * 日付を 1900-01-01 00:00 形式にして返す
 * @param created_at
 * @returns
 */
export const useFormatDate = (created_at: string) => {
  // ゼロ埋め
  const zeroPadding = useCallback((num: number, len: number) => {
    return (Array(len).join('0') + num).slice(-len)
  }, [])

  // 日付フォーマット
  const date = useMemo(() => new Date(created_at), [])
  const year = date.getFullYear()
  const month = zeroPadding(date.getMonth() + 1, 2)
  const day = zeroPadding(date.getDate(), 2)
  const hour = zeroPadding(date.getHours(), 2)
  const min = zeroPadding(date.getMinutes(), 2)
  const datetime = `${year}/${month}/${day} ${hour}:${min}`

  return datetime
}

/**
 * URL から groupId を取得して返す
 * @returns string?
 */
export const useParamGroupId = () => {
  const { groupId } = useParams<{ groupId?: string }>()
  return groupId
}

export const useGroupModal = (groupName: string) => {
  const [isConfirm, setIsConfirm] = useState(false)
  const [isOpen, setIsOpen] = useState(false)
  const [newGroupName, setNewGroupName] = useState('')
  const [isOver, setIsOver] = useState(false)
  const dispatch = useAppDispatch()

  const openModal = useCallback(() => {
    setIsOpen(true)
  }, [])
  const closeModal = useCallback(() => {
    setIsOpen(false)
    setIsConfirm(false)
  }, [])
  const openConfirm = useCallback(() => {
    setIsConfirm(true)
  }, [])
  const closeConfirm = useCallback(() => {
    setIsConfirm(false)
  }, [])

  useEffect(() => {
    if (newGroupName.length <= 15) {
      setIsOver(false)
    } else {
      setIsOver(true)
    }
  }, [newGroupName])

  return [
    { isOpen, isConfirm, isOver, newGroupName },
    { openModal, closeModal, openConfirm, closeConfirm, setNewGroupName },
  ] as const
}
