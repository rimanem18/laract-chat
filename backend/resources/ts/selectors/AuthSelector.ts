import { createSelector } from 'reselect'
import { RootState } from '../app/rootReducer'

export const authSelector = (state: RootState) => state.authSlice

/**
 * name を取得する
 */
export const authNameSelector = createSelector(authSelector, (authSlice) => {
  return authSlice.name
})

/**
 * email を取得する
 */
export const authEmailSelector = createSelector(authSelector, (authSlice) => {
  return authSlice.email
})

/**
 * password を取得する
 */
export const authPasswordSelector = createSelector(
  authSelector,
  (authSlice) => {
    return authSlice.password
  }
)

/**
 * message を取得する
 */
export const authMessageSelector = createSelector(authSelector, (authSlice) => {
  return authSlice.message
})

/**
 * promise を取得する
 */
export const authPromiseSelector = createSelector(authSelector, (authSlice) => {
  return authSlice.promise
})
