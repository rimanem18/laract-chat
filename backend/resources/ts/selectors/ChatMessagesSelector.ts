import { createSelector } from 'reselect'
import { RootState } from '../app/rootReducer'

export const chatMessagesSelector = (state: RootState) =>
  state.chatMessagesSlice

export const selectMessages = {
  /**
   * メッセージのID一覧を取得する
   */
  allIds: createSelector(chatMessagesSelector, (groupsSlice) => {
    return groupsSlice.messages.allIds
  }),
  /**
   * メッセージの実体を取得する
   */
  byId: createSelector(chatMessagesSelector, (chatMessagesSlice) => {
    return chatMessagesSlice.messages.byId
  }),
}

/**
 * promise を取得する
 */
export const chatMessagesPromiseSelector = createSelector(
  chatMessagesSelector,
  (chatMessagesSlice) => {
    return chatMessagesSlice.promise
  }
)
