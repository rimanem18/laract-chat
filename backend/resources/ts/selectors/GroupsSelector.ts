import { createSelector } from 'reselect'
import { RootState } from '../app/rootReducer'

export const groupsSelector = (state: RootState) => state.groupsSlice

/**
 * ids を取得する
 */
export const groupIdsSelector = createSelector(
  groupsSelector,
  (groupsSlice) => {
    return groupsSlice.ids
  }
)

/**
 * entities を取得する
 */
export const groupsEntitiesSelector = createSelector(
  groupsSelector,
  (groupsSlice) => {
    return groupsSlice.entities
  }
)

/**
 * promise を取得する
 */
export const groupsPromiseSelector = createSelector(
  groupsSelector,
  (groupsSlice) => {
    return groupsSlice.promise
  }
)

/**
 * oldestId を取得する
 */
export const groupsOldestIdSelector = createSelector(
  groupsSelector,
  (groupsSlice) => {
    return groupsSlice.oldestId
  }
)
