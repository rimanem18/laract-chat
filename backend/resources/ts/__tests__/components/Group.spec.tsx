import React from 'react'
import '@testing-library/jest-dom'
import { fireEvent, render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { mockState } from '../../app/mockState'
import Group from '../../components/Group'
import { groupNameSelector } from '../../selectors/GroupsSelector'

const mockUseAppDispatch = jest.fn()
const mockUseAppSelector = jest.fn()
jest.mock('../../app/hooks', () => ({
  useAppDispatch:
    () =>
    (...args: any[]) =>
      mockUseAppDispatch(...args),
  useAppSelector:
    () =>
    (...args: any[]) =>
      mockUseAppSelector(...args),
  useGroupName: (id: string) => useGroupNameMock(id),
  useGroupPath: (id: string) => useGroupPathMock(id),
  useParamGroupId: () => useParamGroupIdMock(),
  useGroupModal: () => useGroupModalMock(),
  useUserState: () => useUserStateMock(),
  useGroupsState: () => useGroupsStateMock(),
}))

// Hooks の Mock
const groupState = mockState.groupsSlice
const userState = { roleNumberIds: [1] }
const useParamGroupIdMock = jest.fn().mockReturnValue(1)
const useUserStateMock = jest.fn().mockReturnValue(userState)
const useGroupsStateMock = jest.fn().mockReturnValue(groupState)
let useGroupModalMock = jest.fn()

// 引数のある Hooks
const useGroupNameMock = jest
  .fn()
  .mockImplementation((id: string) => groupState.groups.byId[id].name)
const useGroupPathMock = jest
  .fn()
  .mockImplementation(
    (id: string) => `/groups/${groupState.groups.byId[id].id.toString()}`
  )

const Component = (
  <Router>
    <Group />
  </Router>
)

// テスト開始
const setup = () => {
  const screen = render(Component)
  const groupNames = [
    screen.getByTestId('group1') as HTMLElement,
    screen.getByTestId('group2') as HTMLElement,
  ]
  return {
    groupNames,
    ...screen,
  }
}

describe('Group', () => {
  useGroupModalMock = jest.fn().mockReturnValue([
    {
      isOpen: false,
      isConfirm: false,
      newName: 'Group Name',
    },
    {
      openModal: jest.fn(),
      closeModal: jest.fn(),
      openConfirm: jest.fn(),
      closeConfirm: jest.fn(),
      setNewGroupName: jest.fn(),
    },
  ])

  it('snapshot', () => {
    const tree = renderer.create(Component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('グループ名が表示される', () => {
    const { groupNames } = setup()

    const byId = groupState.groups.byId

    expect(groupNames[0].textContent).toBe(byId.group1.name)
    expect(groupNames[1].textContent).toBe(byId.group2.name)
  })
})
