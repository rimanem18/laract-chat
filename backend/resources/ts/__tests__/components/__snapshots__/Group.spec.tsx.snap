// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group snapshot 1`] = `
Array [
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ControlPointIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.49 2 2 6.49 2 12s4.49 10 10 10 10-4.49 10-10S17.51 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </span>
    新しいグループを追加
  </button>,
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters Mui-selected MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        <div
          className="MuiBox-root css-24ohs5"
          data-testid="group1"
        >
          hello
        </div>
      </span>
    </div>
  </div>,
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        <div
          className="MuiBox-root css-24ohs5"
          data-testid="group2"
        >
          world
        </div>
      </span>
    </div>
  </div>,
]
`;
